### PR TITLE
Add regridding benchmark

### LINKED ISSUE #1556: Regridding

### PR DESCRIPTION
**Pull request by hendrikmakait**
Closes #1556

### CODE REVIEW

**Comment by mrocklin** [comment]
Do we have enough context here to add this to the benchmark post?  If you give me bullet points and an image (if one makes sense) I'm happy to write up words.

**Comment by hendrikmakait** [comment]
@aulemahal: Thanks for the input. I'll add a follow-up issue to look into some of the suggestions for increasing the complexity of this workload.

> IIUC, these benchmarks are more geared towards dask ? 

It is, but it's also aimed at reflecting real workloads. Would more complex methods also result in more complex computations or just in more complex weight generation? 

> We do have some code to make the weights generation in parallel but I would say it is still experimental and of limited scope.

Please let us know if we can help with anything from a Dask perspective.

@mrocklin:
> Do we have enough context here to add this to the benchmark post? If you give me bullet points and an image (if one makes sense) I'm happy to write up words.

I'll whip something up. At first glance, this benchmark seems to do alright; it's mostly an embarrassingly parallel computation. Performance and # of tasks could probably look better but that's already a lot better than some of the other benchmarks.

**Comment by aulemahal** [comment]
> Would more complex methods also result in more complex computations or just in more complex weight generation?

Mostly more complex weight generation, which is totally on the ESMF side, so partly in C/Fortran I think. Maybe two very different grids (curvilinear ones for example) and more complex methods would make for weights with more connected nodes, but I don't think this would affect the computation so much.

**Comment by hendrikmakait** [comment]
@slevang, thanks for the additional input! Would you be interested in contributing a benchmark implemented with `xarray-regrid`?

File: AB_environments/AB_sample.conda.yaml [hunk 1]
```diff
@@ -25,7 +25,7 @@ dependencies:
   - fsspec ==2024.6.1
   - s3fs ==2024.6.1
   - gcsfs ==2024.6.1
-  - pyarrow ==17.0.0
+  - pyarrow ==15.0.0
```
Comment (phofl) [inline AB_environments/AB_sample.conda.yaml:28]: why?

**Review by phofl — COMMENTED** [review]
small comment, otherwise lgtm

File: AB_environments/AB_sample.conda.yaml (hunk 1 above)
Comment (hendrikmakait) [reply AB_environments/AB_sample.conda.yaml:28]: to keep this in sync with `environment.yml`

Comment (phofl) [reply AB_environments/AB_sample.conda.yaml:28]: Ah sorry, unrelated change...

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx

**Pull request merged**